pkg/network/ebpf/c/runtime/http.c

```diff
@@ -191,17 +196,23 @@ int uretprobe__SSL_read(struct pt_regs* ctx) {
     }
 
     void *ssl_ctx = args->ctx;
-    conn_tuple_t *t = tup_from_ssl_ctx(ssl_ctx, pid_tgid);
-    if (t == NULL) {
+    ssl_sock_t *ssl_sock = ssl_sock_from_ssl_ctx(ssl_ctx, pid_tgid, 1);
+    if (ssl_sock == NULL) {
         goto cleanup;
     }
 
-    u32 len = (u32)PT_REGS_RC(ctx);
+    size_t len = (size_t)PT_REGS_RC(ctx);
```

Comment:
It looks like `int` is the correct return type here: https://github.com/openssl/openssl/blob/0fcf2351ecff5db21cba431704e4da631b74904a/include/openssl/ssl.h.in#L1875